Allow mutating input tensor

helunwencser opened the pull request:
To support dynamic kv cache, we need to pass in kv cache as an input tensor and update it inside the model. This PR allows mutating input tensor.

Differential Revision: [D61683366](https://our.internmc.facebook.com/intern/diff/D61683366)

helunwencser commented:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* (to be filled)

helunwencser commented:
@helunwencser has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D61683366).

JacobSzwejbka reviewed (approved):
Please double check that existing models with cache as state still work.

helunwencser committed:
Allow mutating input tensor (#4850)

Summary:

To support dynamic kv cache, we need to pass in kv cache as an input tensor and update it inside the model. This PR allows mutating input tensor.

imported-using-ghimport

Test Plan: Imported from OSS

Reviewed By: JacobSzwejbka

Differential Revision: D61683366

Pulled By: helunwencser
